### PR TITLE
More HTTP/2 tests; test code refactoring

### PR DESCRIPTION
**Pull request by richdougherty**
Started tacking the large `ScalaResultsHandlingSpec` integration test. This required some refactoring to get it working, i.e. filtering of tests, support for `BasicHttpClient`.

I started this work a couple of weeks ago - just wanted to finish it off and get it merged if possible.

For those reviewing - you can ignore most of the changes in the `*Spec` classes. Those changes are just the result of refactoring. To see the real changes look at the `ScalaResultsHandlingEndpointSpec` and at the `play.it.test.*` classes.

### CODE REVIEW

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingWithEndpointsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      }
+    }
+
+    "not add a content length header when none is supplied" in serveResult(
+      Results.Ok.sendEntity(HttpEntity.Streamed(Source(List("abc", "def", "ghi")).map(ByteString.apply), None, None))
+    )
+      .useOkHttp.request("/")
+      // Rich 8-20 Nov 2017: Have seen intermittent failures for Akka HTTP/1.1 over SSL.
+      // The failure is a read timeout in the Sun SSL stack on the OkHttp client side.
+      // Other types of endpoint worked OK.
```
Comment (richdougherty) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingWithEndpointsSpec.scala]: I'll create an issue for this once this PR is merged.

File: framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingWithEndpointsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        response.header(CONTENT_TYPE).toLowerCase(java.util.Locale.ENGLISH) must_== "text/event-stream"
+        response.header(TRANSFER_ENCODING) must_== "chunked"
+        response.header(CONTENT_LENGTH) must beNull
+        response.body.string must_== "data: a\n\ndata: b\n\n"
+      }
+
+    // Rich: Nov 21 2017: The 'ignore cookies' test gives read timeouts
+    // on both Netty and Akka when `serveOk` is replaced with
+    // an action that returns the cookie value. I've left it
+    // as a simple Ok response for now.
```
Comment (richdougherty) [inline framework/src/play-integration-test/src/test/scala/play/it/http/ScalaResultsHandlingWithEndpointsSpec.scala]: I'll also create an issue for this once this is merged.

**Comment by mkurz** [comment]
@richdougherty This needs a rebase.

**Comment by dwijnand** [comment]
Closing as stale, but probably something that would be great to recover and land.

**Pull request closed without merging**